server/storage/memory.go

```diff
@@ -84,11 +90,50 @@ func (st *MemStorage) GetCurrent(gun, role string) (data []byte, err error) {
 func (st *MemStorage) GetChecksum(gun, role, checksum string) (data []byte, err error) {
 	st.lock.Lock()
 	defer st.lock.Unlock()
-	data, ok := st.checksums[gun][checksum]
-	if !ok || len(data) == 0 {
+	v, ok := st.checksums[gun][checksum]
+	if !ok {
 		return nil, ErrNotFound{}
 	}
-	return data, nil
+	return v.data, nil
+}
+
+// GetVersions for MemStorage only returns the latest version
+// as it does not store multiple versions
+func (st *MemStorage) GetVersions(gun, role, start string, numToReturn int) ([][]byte, error) {
+	var (
+		idx, stop int
+		id        = entryKey(gun, role)
+	)
+	if start != "" {
+		for st.tufMeta[id][idx].checksum != start {
+			if idx == len(st.tufMeta[id]) {
```

Comment:
Should this be `len(st.tufMeta[id]) - 1`?  By the time it reaches `len(st.tufMeta[id])`, won't this be an index out of range error?  Perhaps we could add a test for searching an in-memory store for a checksum that doesn't exist?